### ISSUE TITLE
KNeighborsTimeSeriesClassifier in time_series_neighbors.py

### ISSUE THREAD

**Issue by TonyBagnall** [alan-turing-institute/sktime#140]
this classifier requires some attention. To include

1. remove dim_to_use, and use the standard method for throwing exception if passed a multivariate classifeir
2. updated to include TWE
3. I'm not sure why its switching on "metric" in the constructor, seems very clunky. Also, dont call it metric, call it distance
4. Refactor name to something more sensible, use UK spelling for neighbour
5. Add comments for author etc and edit comments and refactor variable names to conform to others

**Comment by mloning** [comment]
For input checks, please use functions from [here](https://github.com/alan-turing-institute/sktime/blob/update_orchestration/sktime/utils/validation/supervised.py).

**Comment by jasonlines** [comment]
just a few notes about this class that I mentioned at the time (put these on slack but to make sure they don't get lost):

> as mentioned at the time, the name, use of metric, etc is all from the sklearn class
>
>this is effectively just a clone of that to allow multivariate data because the original implementation is broken in sklean (aka you can tell it to use multivariate data, but it never passes that flag down to the method that checks the data and it throws an exception because the data is multivariate...)
>
>the majority of the logic is still in sklearn, this was just a dirty hack to make it play nice without needing to rewrite the whole classifier

**Comment by TonyBagnall** [comment]
1) the structure here is fairly complex, I dont really want to mess too much. There seems to be redundant method def check_data_sktime_tsc(X, dim_to_use=0):
2) is easy
3), I think it is because of sklearn use of metric, so cant refactor. 
4), I guess again it is for compatibility with sklearn. Given we say we are a sklearn compatible toolkit, should keep it I suppose. 
5) have done this.

**Comment by jasonlines** [comment]
Cool, just to tidy up I'll close this as I think everyone that Tony mentioned has been resolved. Just a couple of final comments: 

1. the check data method actually reshapes the data ready for the cython measures if the input is a sktime dataframe. So that should be ok as-is
2. done
3. yep
4. yep (begrudgingly!)
5. done
